client/src/pages/Experience/Experience.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from "react";
+import useDeepCompareEffect from "use-deep-compare-effect";
+import SignUpContainer from "components/SignUpContainer";
+import NewExperienceForm from "./NewExperienceForm";
+import AddExperienceButton from "./AddExperienceButton";
+import { availableLanguages } from "utils";
+import { Typography } from "@material-ui/core";
+import axios from "axios";
+
+// TO-DO: ALLOW USER TO CHANGE EXPERIENCE FROM PREVIOUS EXPERIENCE
+const Experience = () => {
+  const [experience, setExperience] = useState([{ C: 1 }]);
+  const [unselectedLanguages, setUnselectedLanguages] = useState(
+    availableLanguages.slice(1)
+  );
+
+  useDeepCompareEffect(() => {
+    // 'knownLanguages' variable is used to have a O(1) lookup when filtering
+    // for unselected languages, preventing nested looping by having to search through
+    // 'experience' state on each pass of filter
+    const knownLanguages = {};
+    experience.forEach(ele => {
+      knownLanguages[Object.keys(ele)[0]] = true;
+    });
+
+    // Make array of languages from copy of availableLanguages array if they are
+    // not already selected
+    const newUnselectedLanguages = availableLanguages.slice().filter(ele => {
```

Comment:
Does slice() have a function here? Array.filter() doesn't mutate the original array, it already returns a new filtered array.